lib/src/display/render_loop.dart

```diff
@@ -62,7 +74,20 @@ class RenderLoop extends RenderLoopBase {
     }
 
     for (int i = 0; i < _stages.length; i++) {
-      _stages[i].materialize(_currentTime, deltaTime);
+      if(_stages[i].renderMode != StageRenderMode.AUTO_DIRTY || juggler.hasAnimatables || _stages[i].juggler.hasAnimatables || _stages[i].dirty){
```

Comment:
That sounds good – and great if you would add `autoInvalidateStage()` to Juggler, too. Because then this pull request would be obsolete, as one could achieve what it does by simply extending `RenderLoopBase`.